test/shard_client.tests.js

```diff
@@ -122,6 +122,28 @@ describe('ShardClient', function() {
     shardClient.take('ip', '10.0.0.2', 1, _.noop);
   });
 
+  it('should invoke GET on the client based on the hash', function(done) {
+    const client = function(params) {
+      this.host = params.host;
+      this.get = function(type, key, callback) {
+        assert.equal(this.host, 'limitd://host-1:9231');
+        assert.equal(type, 'ip');
+        assert.equal(key, '10.0.0.1');
+        assert.isFunction(callback);
+        done();
+      };
+    };
+
+    const ShardClient = ShardClientCtor(client);
+
+    const shardClient = new ShardClient({
+      client,
+      shard: { hosts: [ 'host-1', 'host-2' ] }
+    });
+
+    shardClient.get('ip', '10.0.0.1', _.noop);
```

Comment:
Check that the response is actually what you expect?

Reply:
Added. 👍